src/nodedef.cpp

```diff
@@ -716,6 +717,65 @@ void CNodeDefManager::updateAliases(IItemDefManager *idef)
 	}
 }
 
+#include <fstream>
```

Comment:
Headers should be at the top.  You can add a `// For applyTextureOverrides` comment after it if you like.